sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/resourcehints/ResourceHints.java

```diff
@@ -207,8 +210,25 @@ public int hashCode() {
     }
   }
 
-  /** Sets desired minimal available RAM size to have in transform's execution environment. */
+  /**
+   * Sets desired minimal available RAM size to have in transform's execution environment.
+   *
+   * @param ramBytes specifies a positive RAM size in bytes. A number greater than 1G
+   *     (1_000_000_000L) is typical.
+   */
   public ResourceHints withMinRam(long ramBytes) {
+    if (ramBytes <= 0L) {
+      // TODO(yathu) ignore invalid value as of Beam v2.47.0. throw error in future version.
+      LOG.error(
+          "Encountered invalid non-positive minimum ram hint value {}.\n"
```

Comment:
For both of these errors, the likely cause is passing in an (overflowing) int expression. This may be non-obvious to the user, we should suggest that Longs are required here.

Reply:
added hint in both LOG message